src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -1313,12 +1313,23 @@ static bool _buildComposition(LottieComposition* comp, LottieGroup* parent)
     if (parent->buildDone) return true;
     parent->buildDone = true;
 
+    LottieLayer* prev = nullptr;
     for (auto c = parent->children.begin(); c < parent->children.end(); ++c) {
         auto child = static_cast<LottieLayer*>(*c);
 
         //attach the precomp layer.
         if (child->refId) _buildReference(comp, child);
 
+        if (child->matte.type != CompositeMethod::None) {
+            //no index of the matte layer is provided: the layer above is used as the matte source
+            if (child->mid == -1 && prev) {
```

Comment:
Hmm... it shouldn't take "tp" when prev is nullptr?  It should be something like this?
```
//by front
if (child->mid == -1) {
  if (c == children.begin()) err("sholudn't be happend?!");
  child->matte.target = *(c - 1);
//by index
} else {
   ...
}
```